### PR TITLE
Adjusts Pulse Demon and Prisoner Threat/Pop Requirements

### PR DESCRIPTION
**Pull request by jwhitak**
this is a curse that remains from kanef code

## What this does
Tl;dr math, this is why these antags are kinda rare and only spawn at unusual pops/threats, even with prisoner having triple weight on being triggered.

Flips the population and threat requirement lists for Pulse Demon and Prisoner midrounds/latejoins. Antag datums use a crazy number of confusing lists and variables to try and balance what pops and what threat antags can spawn on. Looking over thing, three outliers existed: Ramblers, Pulse Demons, and Prisoners. After running numbers, it seems that certain lists were put in backwards, messing with when they can spawn. This PR effectively makes them follow the standard rules for spawning in. Low pop needs very high threat now to have Pulse Demons and Prisoners, and not so much threat at high pop. In the worst case, the lists would collide, making certain 'windows' of threat that had to be observed for the antag to spawn in.

For example, during 0-5 player lowpop, Pulse Demons would only spawn in on 5-20 threat exactly, instead of the likely intended 70 threat. Whereas on a packed station of 100 players, a pulse demon would need 70 threat, opposed to the likely intended 5 threat. Note that very specific and unusual 5-20 threat window.

I'm dropping prisoner spawn rate back to standard, since the tripling was implemented because nobody ever saw prisoners. It's likely because of the lists being backwards and needing very specific threat ranges to spawn in.

Ramblers are not changed because their requirements being backwards is a cool niche thing and was likely intended.

Pulse Demon: Threat Requiements (entries by standard pop scaling) from ``5,5,15,15,20,20,20,20,40,70`` to  ``70,40,20,20,20,20,15,15,5,5``.
Prisoners: Reduced triggering weight from triple weight to standard weight.
Prisoners: Population Requirements (entries by standard threat scaling) from ``0,0,10,10,15,15,20,20,20,25`` to ``25,20,20,20,15,15,10,10,0,0``.
Prisoners: Threat Requirements (entries by standard pop scaling) from ``5,5,15,15,20,20,20,20,40,70`` to ``70,40,20,20,20,20,15,15,5,5``.

## Why it's good
Pulse Demons and Prisoners spawn in when they 'should' now.

## Changelog

:cl:
 * tweak: Changed the pop/threat that Prisoners and Pulse Demons can spawn in at.
 
 [gameplay] [gamemode] [balance]

### CODE REVIEW

**Comment by jwhitak** [comment]
Cancel the threat nerf. I forgot that threat was recently nerfed a month ago! The enemy jobs part can stay though.

**Comment by Hinaichigo** [comment]
I wasn't sure if this was deliberate for some reason.. like pulse demons being intended to be more a lowpop antag.

**Comment by jwhitak** [comment]
An antag that shows up ONLY during lower threats, never higher threats, while on deadpop, and can use malf abilities all over the place? I'm not too sure...

File: code/datums/gamemode/dynamic/dynamic_rulesets_midround.dm [hunk 1]
```diff
@@ -914,11 +914,11 @@
 	restricted_from_jobs = list()
 	enemy_jobs = list("Warden","Head of Security")
 	required_enemies = list(1,1,1,1,1,1,1,1,1,1)
-	required_pop = list(0,0,10,10,15,15,20,20,20,25)
+	required_pop = list(25,20,20,20,15,15,10,10,0,0)
 	required_candidates = 1
-	weight = 3
+	weight = 1
```
Comment (gurfan) [inline code/datums/gamemode/dynamic/dynamic_rulesets_midround.dm:919]: BASE_RULESET_WEIGHT is equal to 10. Because the prisoner weight was set to three this made them LESS likely to appear.

**Comment by Hinaichigo** [comment]
PD is the kind of thing where its super easy to discover and kill off at higher pops. It's pretty weak generally and does sort of make more sense as a lowpop antag imo. If we do have everything weighted to higher pop then there might not be much antag variety during lowpop.

File: code/datums/gamemode/dynamic/dynamic_rulesets_midround.dm (hunk 1 above)
Comment (jwhitak) [reply code/datums/gamemode/dynamic/dynamic_rulesets_midround.dm:919]: Initial PR had weight = 1. It was tripled due to Prisoner not firing often. I believe that Prisoner didn't fire often because of the backwards requirements, so I reverted the PR to buff the weight.

If you really think that it should be at 3, I can revert it.

**Comment by jwhitak** [comment]
This is true, PD is better off at mid-low pops.

I do want to say that threat is supposed to measure how much of a danger the antag is to a crew. At deadpop, a pulse demon (assuming at least one engineer has set up power, covered by the PD requirements to spawn in) can cause some serious damage. Higher threat is the cost of this. Lower pops should need higher threat to see the PD spawn as a result.

Conversely, at highpop, a pulse demon is quickly found and caught, so not as much threat there. So a lowered threat requirement makes sense. A pulse demon isn't really 'worth' 70 threat when there's 35+ people on the station, right? So it makes no sense to require it when there's tons of players.

Apply the above to prisoners. A single prisoner coming in on a 1 person round is going to be vastly more dangerous than a single prisoner coming in on a 30 person round. The threat requirements change accordingly.

We let traitors spawn at lowpop with lower threat requirements, but most other antags follow these rules. More threat = higher chance to get non-tator antags. Thus, limited variety. A future PR could reduce the threat costs for other antags at lowpop...

**Comment by Kurfursten** [comment]
Rambler being backwards was intended, and also they are hardcoded to not spawn at high threat
